snapcraft/internal/project_loader/_config.py

```diff
@@ -237,6 +238,8 @@ def __init__(self, project: project.Project) -> None:
             self.data.get("architectures"), project.deb_arch
         )
 
+        conduct_preflight_check(self.project)
```

Comment:
mind adding a comment of why this is done inside project loading instead of earlier in the game like before project loading?

There is one downside to this, for build VMs and lxd, you will only see this after the environment is created (which is also good for other flight checks we might have) but bad for things like checking if multipass or qemu are there.

We might need 2 flight checks, and this one should be done a bit earlier

Reply:
Fixed, should happy much earlier now, good call.